### PR TITLE
SC-62 - fixing accordions on settings to only open one at a time

### PR DESCRIPTION
**Pull request by danbowles**

### CODE REVIEW

File: app/components/ChartSettings/index.js [hunk 1]
```diff
@@ -21,14 +22,25 @@ class ChartSettings extends Component {
     return -1 !== this.state.modules.indexOf(name);
   }
 
+  _toggleModuleCallback(expanded, name) {
+    if (expanded) {
+      this.setState({
+        activeSettings: name,
```
Comment (meddy) [inline app/components/ChartSettings/index.js]: The "settings" are referenced as modules everywhere else. For consistency I would do that here as well.

Also, "expanded" over active.

`expandedModule`

File: app/components/ChartSettings/index.js [hunk 2]
```diff
@@ -21,14 +22,25 @@ class ChartSettings extends Component {
...
   _renderModule(name) {
     if (!this._hasModule(name)) {
       return null;
     }
 
     // Setup props, handling special cases for Metadata and ColorPalette
     const moduleProps = update({}, {
-      defaultExpand: { $set: this._shouldDefaultExpand() },
+      defaultExpand: { $set: this._shouldDefaultExpand(name) },
+      toggleCallback: {
```
Comment (meddy) [inline app/components/ChartSettings/index.js]: The typical naming convention for prop callback is onEventName, ie. onClick, onChange, etc..

`onToggle` - prop name
`onModuleToggle` - parent component handler

File: app/components/ChartSettings/index.js [hunk 3]
```diff
@@ -21,14 +22,25 @@ class ChartSettings extends Component {
...
+
   _renderModule(name) {
     if (!this._hasModule(name)) {
       return null;
     }
 
     // Setup props, handling special cases for Metadata and ColorPalette
     const moduleProps = update({}, {
-      defaultExpand: { $set: this._shouldDefaultExpand() },
+      defaultExpand: { $set: this._shouldDefaultExpand(name) },
```
Comment (meddy) [inline app/components/ChartSettings/index.js]: I think it makes sense to use this prop for both open by default and open because toggled, like you set up, but "default" doesn't really make sense anymore. Maybe just `expanded`?

File: app/components/ChartSettings/index.js [hunk 4]
```diff
@@ -52,19 +64,16 @@ class ChartSettings extends Component {
...
   render() {
     return (
       <div>
         <div>
-          {this._renderModule('XAxis')}
-          {this._renderModule('YAxis')}
-          {this._renderModule('Legend')}
-          {this._renderModule('Metadata')}
-          {this._renderModule('ColorPalette')}
+          {this.state.modules.map((name) => this._renderModule(name))}
```
Comment (meddy) [inline app/components/ChartSettings/index.js]: If you iterate over components, you have to provide a key prop.

Comment (danbowles) [inline app/components/ChartSettings/index.js]: I'm opting to revert this change for now, but Guessing you are referring to something like this:

```const listItems = numbers.map((number) =>
  <li key={number.toString()}>
    {number}
  </li>
);```
https://guides.github.com/features/mastering-markdown/

**Comment by joshkadis** [comment]
we'll cross this bridge if we come to it
